lite2/client_test.go

```diff
@@ -987,3 +987,49 @@ func TestProvider_TrustedHeaderFetchesMissingHeader(t *testing.T) {
 	assert.Error(t, err)
 	assert.Nil(t, h)
 }
+
+func TestNew_Client_From_Trusted_Store(t *testing.T) {
+	const (
+		chainID = "TestNew_Client_From_Trusted_Store"
+	)
+
+	var (
+		keys = genPrivKeys(4)
+		// 20, 30, 40, 50 - the first 3 don't have 2/3, the last 3 do!
+		vals     = keys.ToValidators(20, 10)
+		bTime, _ = time.Parse(time.RFC3339, "2006-01-02T15:04:05Z")
+		header   = keys.GenSignedHeader(chainID, 1, bTime, nil, vals, vals,
+			[]byte("app_hash"), []byte("cons_hash"), []byte("results_hash"), 0, len(keys))
+		primary = mockp.New(
+			chainID,
+			map[int64]*types.SignedHeader{
+				1: header,
+			},
+			map[int64]*types.ValidatorSet{
+				1: vals,
+				2: vals,
+			},
+		)
+	)
+
+	// 1) Initiate DB and fill with a "trusted" header
+	db := dbs.New(dbm.NewMemDB(), chainID)
+	err := db.SaveSignedHeaderAndNextValidatorSet(header, vals)
+	assert.NoError(t, err)
+
+	// 2) Initialize Lite Client from Trusted Store
+	c, err := NewClientFromTrustedStore(
+		chainID,
+		1*time.Hour,
+		primary,
+		[]provider.Provider{primary},
+		db,
+	)
+	require.NoError(t, err)
+
+	// 3) Check header exists through the lite clients eyes
+	h, err := c.TrustedHeader(1, bTime.Add(1*time.Second))
```

Comment:
how do you know it wasn't fetched from primary?

Reply:
does the lite client automatically update from the primary on start up?

Reply:
no, but TrustedHeader can fetch missing headers, so how do you now they are not coming from primary?

Reply:
but trusted header would return an error if the database was empty: "no headers exist"

Reply:
https://github.com/tendermint/tendermint/pull/4374/commits/3bc2b1c5e20078e6b7a00c14dee6e720fbbc3ab9

Reply:
You're right. Anyway, it never a bad idea to double check 😄